File: coko/com.elster.jupiter.customtask.rest/src/main/java/com/elster/jupiter/customtask/rest/impl/TaskInfo.java
/*
 * Copyright (c) 2017 by Honeywell International Inc. All Rights Reserved
 */

package com.elster.jupiter.customtask.rest.impl;

import com.elster.jupiter.tasks.RecurrentTask;

public class TaskInfo {

    public Long id;
    public String name;
    public String application;
    public String type;

    public TaskInfo() {

    }

    public static TaskInfo from(RecurrentTask recurrentTask) {
        TaskInfo info = new TaskInfo();

        info.id = recurrentTask.getId();
        info.name = recurrentTask.getName();
        info.application = recurrentTask.getApplication();
        info.type = "";

        return info;
    }

}

File: copl/com.elster.jupiter.appserver.extjs/src/main/web/js/appserver/src/model/TaskInfo.js
/*
 * Copyright (c) 2017 by Honeywell International Inc. All Rights Reserved
 */

Ext.define('Apr.model.TaskInfo', {
    extend: 'Ext.data.Model',
    fields: [
        'id', 'application', 'name', 'type',
        {
            name: 'displayType',
            persist: false
        },
        {
            name: 'displayName',
            persist: false,
            convert: function (value, rec) {
                return Ext.String.format(Uni.I18n.translate('task.taskType', 'APR', '{0} ({1})', [rec.get('name'), rec.get('displayType')]));
            }
        }
    ]
});
